cli/buildengine.ts

```diff
@@ -118,6 +118,15 @@ export function buildHexAsync(buildEngine: BuildEngine, mainPkg: pxt.MainPackage
     let allFiles = U.clone(extInfo.generatedFiles)
     U.jsonCopyFrom(allFiles, extInfo.extensionFiles)
 
+    for (let f of nodeutil.allFiles(buildEngine.buildPath)) {
+        let bn = f.slice(buildEngine.buildPath.length)
+        bn = bn.replace(/\\/g, "/").replace(/^\//, "/")
+        if (U.startsWith(bn, "/source/") && !allFiles[bn]) {
+            pxt.log("removing stale " + bn)
+            fs.unlinkSync(f)
```

Comment:
I really only want to nuke files, and also not update timestamps of the ones that didn't change (to avoid excessive rebuilds)